### ISSUE TITLE
Appium is Unable to Find iOS Accessibility Identifiers on iOS 15

### ISSUE THREAD

**Issue by T61Z8T3** [appium/appium#16235]
## The problem

After running our test cases on os_version 15, I faced the following issue:
Appium is unable to find the iOS accessibility IDs on some of the UITableView. The locators could be found before (on os version 14 or 13 ).
A screenshot of our app on iPhone 12 os version 14 which is fine and the elements can be found: 
![image](https://user-images.githubusercontent.com/78351604/145466840-43f2ab80-4953-41b9-8253-5e5552f1e7da.png)

A screenshot of our app with same capabilities, the only change is here the device is iPhone 11 Pro with os version of 15: 
![image](https://user-images.githubusercontent.com/78351604/145467800-ae44e457-a633-40cd-82ad-4742e06ea85e.png)

## Environment

* Appium version (or git revision) that exhibits the issue: Appium Desktop version <1.22.1>
* Last Appium version that did not exhibit the issue (if applicable):  The Appium version is the same
* Desktop OS/version used to run Appium: Mac version 11.6.1 (20G224) - using Appium inspector 2021.9.2 
* Node.js version (unless using Appium.app|exe): 14.15.0
* Npm or Yarn package manager: npm package manager
* Mobile platform/version under test: iPhone12 , os_version: 14 and iPhone pro 11 os_version: 15
* emulator/simulator: running on Browserstack
* Appium Desktop version <1.22.1>

## Details

If necessary, describe the problem you have been experiencing in more detail.

## Link to Appium logs

Create a [GIST](https://gist.github.com) which is a paste of your _full_ Appium logs, and link them here.
Do _NOT_ paste your full Appium logs here, as it will make this issue very long and hard to read!
If you are reporting a bug, _always_ include Appium logs!

**Comment by KazuCocoa** [comment]
What is the result of https://github.com/appium/appium-xcuitest-driver#mobile-source as `format: description`?

**Comment by T61Z8T3** [comment]
> What is the result of https://github.com/appium/appium-xcuitest-driver#mobile-source as `format: description`?

In the iOS side we already set accessibilityIdentifier and  .isAccessibilityElement = true for each element.

**Comment by KazuCocoa** [comment]
Could you share the result? We wondered if the vanilla debug information had your expected elements.

**Comment by T61Z8T3** [comment]
Can you please explain what do you mean by vanilla debug?

**Comment by KazuCocoa** [comment]
I meant by XCTest debug description that was available via https://github.com/appium/appium/issues/16235#issuecomment-990246872

**Comment by T61Z8T3** [comment]
[iphone12-OS-14-OK.txt](https://github.com/appium/appium/files/7713556/iphone12-OS-14-OK.txt)
[iphone11Pro-OS-15-NotOK.txt](https://github.com/appium/appium/files/7713557/iphone11Pro-OS-15-NotOK.txt)

**Comment by KazuCocoa** [comment]
Thank you.
No `swipeHintLabel` in the iOS 15 one. It means XCTest did not recognize them. Then, Appium cannot find it unfortunately since XCTest framework by Apple itself did not detect it. Only Apple can fix it.

**Comment by KazuCocoa** [comment]
I don't know if this helps, but you could try out tuning accessibility preference as https://appium.io/docs/en/drivers/ios-xcuitest/#accessibility-preferences-fine-tuning

**Comment by vkochar** [comment]
Seeing similar issue. Was there a workaround implemented in appium for this to work on iOS 15 or anything I can do to get around this issue? Thanks

**Comment by KazuCocoa** [comment]
No, if your element did not appear even via https://github.com/appium/appium/issues/16235#issuecomment-990246872 , current Appium XCUITest driver can do nothing unfortunately. Perhaps you could report it to Apple.
